src/core/common/message.cpp

```diff
@@ -103,22 +104,25 @@ Buffer *MessagePool::NewBuffer(Message::Priority aPriority)
 {
     Buffer *buffer = nullptr;
 
-    SuccessOrExit(ReclaimBuffers(1, aPriority));
-
-#if OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
-
-    buffer = static_cast<Buffer *>(otPlatMessagePoolNew(&GetInstance()));
-
+    while ((
```

Comment:
Why using `while` here instead `if`? It seems that the line `SuccessOrExit(ReclaimBuffers(aPriority));` won't execute for a second time. And even it does, the result won't change, then an infinite loop occurs.